processor/src/main/java/org/mapstruct/ap/internal/processor/MethodRetrievalProcessor.java

```diff
@@ -512,9 +513,11 @@ private boolean checkParameterAndReturnType(ExecutableElement method, List<Param
                 return false;
         }
 
-        Type parameterType = sourceParameters.get( 0 ).getType();
+        Optional<Type> parameterType =
```

Comment:
I rember that we had discussions some time ago  on whether you should use optionals this way 😄 . 

Anyway, woudn't it be better to wrap line 516 to 567 in an 'if' statement like this:

```java
if souceParameters.size()==1){
 : 

}
```
to make it more explicit?

Reply:
Zooming further out, this seams to be very much geared towards checking certain types of mapping methods. I'm thinking: should that be checked here really? Or at the moment when we really try to do the iterable-to-iterable mapping in the 'MappingCreationProcessor' ?  I guess it's some kind of fail-fast. 

And just curious for that matter.. what happens if we combine a bean with an interable as both source? Is it than automatically recognized as bean-mapping?

Reply:
haha, you are right. We had a discussion about the optionals used in this way. I'll try to extract something better.

And yes this is done here for fail fast. I don't want to start moving things around as it might make some things more complicated 😄.

>And just curious for that matter.. what happens if we combine a bean with an interable as both source? Is it than automatically recognized as bean-mapping?

Yes it will be recognized as bean mapping. The other types of mappings are mappings with only a single source parameter.